src/hal/sensor/sensor.cpp

```diff
@@ -0,0 +1,80 @@
+
+#include "../../hal/hal.h"
+#include "../../hal/sensor/sensor.h"
+
+// Sensor pin defines
+#define FLOW_SENSE_PIN     A0
+#define PRESSURE_SENSE_PIN A1
+
+
+// The following function initializes Timer 5 to do pressure and flow sensor data collection
+int16_t sensorHalInit(void)
+{
+  // Sensor timer
+  TCCR5A=0; // set timer control registers
+  TCCR5B=0x0A;  // set timer to clear timer on compare (CTC), and prescaler to divide by 8
+  TCNT5=0;  // initialize counter to 0
+  OCR5A= 1000;  // Set interrupt to 2khz
+  TIMSK5 |= 0x02; // enable timer compare interrupt
+
```

Comment:
Since we want to sample the sensors at 40Hz each (is that fast enough to oversample?), why not just run the timer at 80Hz (change prescaler at /64, set OCR5A to 3125, changing `timer5Counter` to a bit flip instead)? saves many unneeded interrupts.

Reply:
Also; can we use the same style used in the motor control for setting bits, I know its verbose but its somewhat easier to debug without having to refer to the datasheet for the bit position of each setting)?

Reply:
The flow sensor results get integrated into the volume pseudo-sensor. Initial tests showed good repeatability at 100Hz or 10ms sampling intervals. The lower the flow sampling frequency is - the greater error/noise we'll be getting as volume measurements. Unless we are cycles constrained (where?) I'd suggest we keep reading flow and integrating the volume at 100Hz.

Reply:
I think I miscalculated the timer speed sorry; it is set to 2KHz and the timer isr further divides that by 20. I was just suggesting to make the timer run at 200Hz directly and divide the timer isr by 2 instead, saving 18 unneeded isrs. I somehow got to a sample rate of 40Hz; and not Im not sure what math got me there. So, make `OCR5A = 10000` and make the counter a boolean flag.